### PR TITLE
Support delay option

### PR DESCRIPTION
**Pull request by rv2e**
The goal is to be able to delay the request. Sometimes requests take long time and it would be great to set the same latency in the mock to see how the application handles this latency.

### CODE REVIEW

File: lib/superagent-mock.js [hunk 1]
```diff
@@ -142,7 +142,7 @@ function mock (superagent, config, logger) {
       }
     }
 
-    fn(error, response);
```
Comment (fdubost) [inline lib/superagent-mock.js]: Even if it's longer, I would prefer to not run `setTimeout` if there is no delay 😉 
```js
if (context.delay) {
    setTimeout(function() {
        fn(error, response);
    }, context.delay);
} else {
    fn(error, response);
}
```

**Comment by fdubost** [comment]
And some doc in the README too 🎁

**Comment by rv2e** [comment]
Sure, I'll try take time to complete the README and the tests this week-end 👍

**Comment by elisherer** [comment]
This is amazing! I didn't notice your pull request and made almost the same thing.
[__Add support for asynchronously delaying mocked responses__ #52]

I see you missed adding unit tests for the code. 
Also I checked if context.delay is a number to even allow 0 as a timeout.
Maybe you could go over my code and see what you can add from it, and I will cancel my PR.

Great minds think alike 😄

File: lib/superagent-mock.js [hunk 2]
```diff
@@ -142,7 +141,7 @@ function mock (superagent, config, logger) {
...
+    setTimeout(function () { fn(error, response); }, context.delay || 0);
```
Comment (elisherer) [inline lib/superagent-mock.js]: You are referring to context while it wasn't declared in that scope (it was declared under the try block scope above). Needs getting it out before the try-catch block.

File: lib/superagent-mock.js (hunk 1 above)
Comment (elisherer) [inline lib/superagent-mock.js]: Will do 
```js
if (typeof(context.delay) === 'number') {
```
instead and then even support doing setTimeout(..., 0)

**Comment by rv2e** [comment]
Maybe we could use your PR instead of mine since they are almost the same but you've already written some tests and docs 👍 
Should I close this PR then and make comments on your PR?

**Comment by elisherer** [comment]
Sure, make your comments on mine

**Pull request closed without merging**